wallet/tutorials/javascript-dapp-simple.md

```diff
@@ -0,0 +1,372 @@
+Then, edit `package.json` to include a build script:
+
+```json
+"scripts": {
+    "build": "webpack"
+}
+```
+
+When you run `npm run build` from your terminal, Webpack bundles the content in `src/index.js` into `dist/main.js`, which is linked in `index.html`.
+
+### 2. Detect MetaMask
+
+You can use the `@metamask/detect-provider` module to detect the MetaMask Ethereum provider.
+
+#### Install `@metamask/detect-provider`
+
+Use [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) to install
+`@metamask/detect-provider` in your project directory:
+
+```bash
+npm i @metamask/detect-provider
+```
+
+#### Detect the MetaMask provider
+
+The following code example demonstrates how to detect the MetaMask provider in your dapp using `@metamask/detect-provider`:
+
+```jsx title="src/index.js"
+import detectEthereumProvider from "@metamask/detect-provider";
+
+async function setup() {
+    const provider = await detectEthereumProvider();
+
+    if (provider && provider === window.ethereum) {
+        console.log('MetaMask is available!');
+        startApp(provider); // Initialize your dapp with MetaMask.
+    } else {
+        console.log("Please install MetaMask!");
+    }
+}
+
+function startApp(provider) {
+    if (provider !== window.ethereum) {
+        console.error("Do you have multiple wallets installed?");
+    }
+}
+
+window.addEventListener('load', setup);
+```
+#### Serve the project
+
+To test and verify your project works with MetaMask, serve your project through a local server. 
+If using VS Code, right-click the `index.html` file, and select **Open with Live Server**. 
+
+Alternatively, you can install `http-server` using the following command:
+
+```bash
+npm install -g http-server
+``` 
+Then navigate to the project directory, and run `http-server`. 
+
+Open your browser and go to the provided local server URL (typically http://127.0.0.1:8080).
+
+### 3. Determine a user's connected network
+
+Each network has a unique Chain ID. 
+Determining the connected network requires querying this Chain ID.
+You can track the network’s [chain ID](../../how-to/manage-networks/detect-network/#chain-ids.md) to identify which network a user is connected to.
+
+Use the [`eth_chainId`](/wallet/reference/eth_chainId)
+RPC method to detect the chain ID of the user's current network.
+Listen to the [`chainChanged`](/wallet/reference/provider-api/#chainchanged) provider event to
+detect when the user changes networks.
+
+For example, the following code detects a user's network and when the user changes networks:
+
+```jsx title="index.js"
+const chainId = await window.ethereum 
+    .request({ method: "eth_chainId" });
+
+window.ethereum
+    .on("chainChanged", handleChainChanged);
+
+function handleChainChanged(chainId) {
+    // We recommend reloading the page, unless you must do otherwise.
+    window.location.reload();
+}
+```
+For more information on Chain IDs, see [Detect a user's network](../how-to/connect/detect-metamask.md)
```

Comment:
````suggestion
```js title="index.js"
const chainId = await window.ethereum 
  .request({ method: "eth_chainId" });

window.ethereum
  .on("chainChanged", handleChainChanged);

function handleChainChanged(chainId) {
  // We recommend reloading the page, unless you must do otherwise.
  window.location.reload();
}
```
````